src/Generators/MigrationGenerator.php

```diff
@@ -304,13 +303,49 @@ protected function buildPivotTableDefinition(array $segments): string
                 $this->hasForeignKeyConstraints = true;
                 $definition .= $this->buildForeignKey($foreign, $on, 'id') . ';' . PHP_EOL;
             } else {
-                $definition .= self::INDENT . '$table->foreignId(\'' . $foreign . '\');' . PHP_EOL;
+                $definition .= $this->generateForeignKeyDefinition($segment, $foreign, $models);
             }
         }
 
         return trim($definition);
     }
 
+    /**
+     * Generates the foreign key definition for a pivot table.
+     *
+     * This function generates the foreign key definition for a pivot table in a migration file.
+     * It checks if the model exists and its name matches the pivot table segment. If it does,
+     * it determines the data type of the primary key and appends the appropriate method call
+     * to the `$definition` string. If the model does not exist or its name does not match the
+     * pivot table segment, it defaults to appending `$table->foreignId(\'' . $foreignKeyColumnName . '\');`
+     * to the `$definition` string. The function then returns the `$definition` string.
+     *
+     * @param  string  $pivotTableSegment The segment of the pivot table. e.g 'dive_job' it would be 'Dive' or 'Job'.
+     * @param  string  $foreignKeyColumnName The name of the foreign key column. e.g 'dive_id' or 'job_id'.
+     * @param  array  $models An array of models. e.g ['Dive' => $diveModel, 'Job' => $jobModel].
+     * @return string The foreign key definition. e.g '$table->foreignUlid('dive_id');'
+     */
+    protected function generateForeignKeyDefinition(string $pivotTableSegment, string $foreignKeyColumnName, array $models = []): string
```

Comment:
Not sure I understand the reason for a `foreach` here. The original code was a single line, why do we need to look now?

Reply:
The original [code](https://github.com/laravel-shift/blueprint/blob/master/src/Generators/MigrationGenerator.php#L297) had the `foreach`. This function replaces this [line](https://github.com/laravel-shift/blueprint/blob/master/src/Generators/MigrationGenerator.php#L307) in the original function

Reply:
I think this has to do with the pivot table needing to review other models for their `id` column type.